packages/core-ethereum/src/index.ts

```diff
@@ -74,14 +74,23 @@ export default class HoprEthereum extends EventEmitter {
       this.options?.maxConfirmations ?? CONFIRMATIONS,
       INDEXER_BLOCK_RANGE
     )
-    this.status = 'CREATING'
-    log('Ready to call singleton wrapper', ChainWrapperSingleton)
-    ChainWrapperSingleton.create(this.options, this.privateKey).then((chain: ChainWrapper) => {
-      // Debug log used in e2e integration tests, please don't change
-      log(`connector chain created`)
-      this.status = 'CREATED'
-      this.chain = chain
-    })
+    // In some cases, we want to make sure the chain within the connector is not triggered
+    // automatically but instead via an event. This is the case for `hoprd`, where we need
+    // to get notified after ther chain was properly created, and we can't get setup the
+    // listeners before the node was actually created.
+    if (automaticChainCreation) {
+      this.createChain()
+    } else {
+      this.on('connector:create', this.createChain)
```

Comment:
```suggestion
      this.once('connector:create', this.createChain)
```

I suspect we only want this to be triggered once within the node's lifecycle?

Reply:
That's correct, thanks for the suggestion ☺️